codex/node.nim

```diff
@@ -213,6 +219,28 @@ proc fetchBatched*(
   let iter = Iter[int].new(0 ..< manifest.blocksCount)
   self.fetchBatched(manifest.treeCid, iter, batchSize, onBatch, fetchLocal)
 
+proc fetchDatasetAsync*(
+    self: CodexNodeRef, manifest: Manifest, fetchLocal = true
+): Future[void] {.async: (raises: []).} =
+  ## Asynchronously fetch a dataset in the background.
+  ## This task will be tracked and cleaned up on node shutdown.
+  ##
+  try:
+    if err =? (
+      await self.fetchBatched(
+        manifest = manifest, batchSize = DefaultFetchBatch, fetchLocal = fetchLocal
+      )
+    ).errorOption:
+      error "Unable to fetch blocks", err = err.msg
+  except CatchableError as exc:
+    error "Error fetching blocks", exc = exc.msg
```

Comment:
I don't think we should log an error when cancelled.
```suggestion
  except CancelledError as exc:
    discard
  except CatchableError as exc:
    error "Error fetching blocks", exc = exc.msg
```

Reply:
Why?

Reply:
Because cancellation isn't really an error. A cancellation would happen on stop, which is expected, then we'd see an error in the logs

Reply:
Yeah, changing to trace would be better, I still like to have the log message is useful, it can be cleaned up once the code is more stable.